### PR TITLE
Add docs for default iPXE boot setting

### PR DESCRIPTION
**Pull request by anhou**
Following docs PR for https://github.com/RackHD/on-http/pull/186 and https://github.com/RackHD/on-dhcp-proxy/pull/29

@RackHD/corecommitters @RackHD/rackhd_dev

### CODE REVIEW

File: docs/rackhd/boot_settings.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
+Customize Default iPXE Boot Setting
+---------------------------
+
+When a node's NIC's PXE boot is set to the first boot order in BIOS, default boot settings could be customized by iPXE in RackHD, node will boot with default boot settings after node is discovered by RackHD.
```
Comment (heckj) [inline docs/rackhd/boot_settings.rst]: Hey Andrew,

How about this for slightly altered english:
> A compute server's BIOS can be set to always PXE boot using the BIOS boot order to always network boot. The default RackHD response when no workflow is operating is to do nothing - normally falling through to the next item in the BIOS boot order. RackHD can also be configured with a default iPXE script to provide boot instructions when no workflow is operational against the node.

File: docs/rackhd/boot_settings.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Default iPXE Boot To Customized OS in RAM
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In some use cases, node wants to boot customized kernel and initrd to OS in RAM, not boot into a installed distribution OS in disk. This functionality could be provided by `PATCH` **bootSettings** to a node by API command:
```
Comment (heckj) [inline docs/rackhd/boot_settings.rst]: > To configure RackHD to provide a custom iPXE response to a node outside of a workflow running, such as booting a customized kernel and initrd, you can do so by providing configuration to the Node resource in RackHD. This functionality can be enabled by using a `PATCH` REST API call adding **bootSettings** to a node.

**Comment by heckj** [comment]
I made some suggestions for different grammar to help explain better, hopefully it's of use. 👍

**Comment by anhou** [comment]
Thanks @heckj  the descriptions are more better than mine, they're updated.

**Comment by yyscamper** [comment]
:+1:

**Pull request merged**